### PR TITLE
Improve Python3 Compatibility

### PR DESCRIPTION
**Pull request by antiagainst**
* Use print_function from __future__ to tackle the print syntax
  change.
* In Python3 str and bytes are no longer the same thing. Need
  proper decode.

Counterpart for https://github.com/KhronosGroup/SPIRV-Tools/issues/144

### CODE REVIEW

**Comment by antiagainst** [comment]
Now it should build cleanly with Python3. But sadly, when testing, `glslc_test_framework.py` and related things are still not Python3 compatible. Will be the next thing to fix.

File: utils/update_build_version.py [hunk 1]
```diff
@@ -55,21 +57,26 @@ def describe(dir):
...
-        print 'usage: {0} <shaderc_dir> <spirv-tools_dir> <glslang_dir>'.format(sys.argv[0])
+        print(
+            'usage: {0} <shaderc_dir> <spirv-tools_dir> <glslang_dir>'.format(sys.argv[0]))
         sys.exit(1)
 
-    new_content = ('"shaderc ' + describe(sys.argv[1]).replace('"', '\\"') + '\\n"\n' +
-                   '"spirv-tools ' + describe(sys.argv[2]).replace('"', '\\"') + '\\n"\n' +
-                   '"glslang ' + describe(sys.argv[3]).replace('"', '\\"') + '\\n"\n')
+    tags = [
+        describe(p).decode('ascii').replace('"', '\\"')
```
Comment (dekimir) [inline utils/update_build_version.py]: Why is `decode('ascii')` necessary here?

Comment (antiagainst) [inline utils/update_build_version.py]: In Python2, `str` is `bytes`, but not in Python3.

https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data

Here, `Popen.communicate()` returns `bytes`, which in Python3 is binary data, cannot be directly `+`'d (or invoking `str`-related methods) to `str`, which is text data.

Comment (dekimir) [inline utils/update_build_version.py]: I think there should be a comment explaining this, as it's non-obvious.

Comment (antiagainst) [inline utils/update_build_version.py]: Done.

**Comment by dekimir** [comment]
+2

**Pull request merged**